cmd/shortener/handlers.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"shorturl/internal/config"
+)
+
+func generateURL(res http.ResponseWriter, req *http.Request) {
+	if req.URL.Path != "/" {
+		http.Error(res, "Wrong request path", http.StatusBadRequest)
+		return
+	}
+
+	url, err := io.ReadAll(req.Body)
+	if err != nil {
+		http.Error(res, "Can't read body", http.StatusBadRequest)
+		return
+	}
+
+	hash := getHashFromURL(url)
+	err = storageURLs.Set(hash, string(url))
+	if err != nil {
+		http.Error(res, err.Error(), http.StatusBadRequest)
```

Comment:
```suggestion
		http.Error(res, err.Error(), http.StatusInternalServerError)
```
Это все-таки ошибка сервера, а не пользовательского ввода.

Reply:
Fixed.